wire-ios/Wire-iOS Tests/DeviceManagement/MockLogger.swift

```diff
@@ -0,0 +1,55 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+final class MockLogger: LoggerProtocol {
```

Comment:
question: why do we need mockLogger?

Reply:
@theswiftway please reply

Reply:
Right now we don't have unit tests for checking if logging was done when an error is thrown. This mock was setup to do that. Alternatively, Sourcery could be used.